RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -1013,11 +1039,19 @@ namespace {
     int getPhiTensorIndex(const CellIndex& cellIndex) const { return cellIndex.phi + maxPhiIndex(); }
 
     bool tryGetCellIndex(double deltaEta, double deltaPhi, CellIndex& cellIndex) const {
-      static auto getCellIndex = [](double x, double maxX, double size, int& index) {
+      static auto getCellIndex = [this](double x, double maxX, double size, int& index) {
```

Comment:
```suggestion
      const auto getCellIndex = [this](double x, double maxX, double size, int& index) {
```
to match #32838

Reply:
added backport of #32838 and updated PR description accordingly